src/figuresClasses.ts

```diff
@@ -1,11 +1,78 @@
-export interface Figure {}
+type Color = `red` | `green` | `blue`;
+type Shape = `triangle` | `circle` | `rectangle`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('your error message');
+    }
+
+    const sortedTypes = [a, b, c].sort((arg1, arg2) => arg1 - arg2);
+
+    if (!(sortedTypes[0] + sortedTypes[1] > sortedTypes[2])) {
+      throw new Error('your error message');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message.
![image](https://github.com/mate-academy/ts_figures_classes/assets/63243136/5f1934ad-2188-40d2-9b76-c434dced8549)

Fix it in all places